Mathlib/RingTheory/Localization/LocalizationLocalization.lean

```diff
@@ -174,7 +174,27 @@ theorem localization_isScalarTower_of_submonoid_le (M N : Submonoid R) (h : M 
   letI := localizationAlgebraOfSubmonoidLe S T M N h
   IsScalarTower.of_algebraMap_eq' (IsLocalization.lift_comp _).symm
 
-noncomputable instance (x : Ideal R) [H : x.IsPrime] [IsDomain R] :
+/-- Let `R` and `S` be ring with fields of fraction `K` and `L`. Suppose that `S` is an `R` algebra
+and let `Rₚ` be the localization of `R` at a submonoid `M`. Let `Sₚ` be the induced localization
+of `S`. Then `IsScalarTower Rₚ Sₚ L`.
+
+The actual statement only requires minimal assumptions on `R`, `Rₚ`, `S`, `Sₚ`, `K` and `L`, but
+the above is the main application. -/
+theorem localization_localization_isScalarTower (Rₚ Sₚ K L : Type*) [CommRing Rₚ]
```

Comment:
This lemma looks very strange to me. 
We should have the following four lemmas given an algebra tower `A -> B -> C -> D`:
1. `IsScalarTower A B D + IsScalarTower B C D + IsScalarTower A B C => IsScalarTower A C D`
2. `IsScalarTower A C D + IsScalarTower B C D + IsScalarTower A B C => IsScalarTower A B D`
3. `IsScalarTower A B D + IsScalarTower A B D + IsScalarTower A B C => IsScalarTower B C D` if `B/A` is epi, e.g. a localization.
4. `IsScalarTower A C D + IsScalarTower A B D + IsScalarTower B C D => IsScalarTower A B C` if `D/C` is mono, i.e. injective (`FaithfulSMul`)

Then this is just some successive application of these four (1 on `RSSₚL` to get `IsScalarTower R Sₚ L`, 2 on `RRₚKL` to get `IsScalarTower R Rₚ L`, and 3 on `RRₚSₚL` to get `IsScalarTower Rₚ Sₚ L`)

Reply:
Well, it's actually so specific that it's unlikely it will ever be used, so I just removed it and proved the instance by hand (the proof is essentially what you suggest, it just a repeated application of `IsScalarTower.algebraMap_eq`).

Reply:
I think we should add the lemmas I mentioned though, or else people will need to go through the rewrite maze whenever they need this.